cmd/frontend/graphqlbackend/external_accounts.go

```diff
@@ -136,5 +145,111 @@ func (r *schemaResolver) DeleteExternalAccount(ctx context.Context, args *struct
 		return nil, err
 	}
 
+	permssync.SchedulePermsSync(ctx, r.logger, r.db, protocol.PermsSyncRequest{
+		UserIDs: []int32{account.UserID},
+	})
+
 	return &EmptyResponse{}, nil
 }
+
+func (r *schemaResolver) AddExternalAccount(ctx context.Context, args *struct {
+	ServiceType    string
+	ServiceID      string
+	AccountDetails string
+}) (*EmptyResponse, error) {
+	user, err := auth.CurrentUser(ctx, r.db)
+	if err != nil {
+		return nil, err
+	}
+	if user == nil {
+		return nil, errors.New("not authenticated")
+	}
```

Comment:
Since you only need the `user.ID` I think you can save yourself doing the DB query again:

```suggestion
	a := actor.FromContext(ctx)
	if !a.IsAuthenticated() {
		return nil, auth.ErrNotAuthenticated
	}
```

And then use `a.UID` instead of `user.ID`.